samcli/lib/deploy/deployer.py

```diff
@@ -72,13 +73,18 @@
 
 OUTPUTS_TABLE_HEADER_NAME = "CloudFormation outputs from deployed stack"
 
+# 500ms of sleep time between stack checks and describe stack events.
+DEFAULT_CLIENT_SLEEP = 0.5
+
 
 class Deployer:
     def __init__(self, cloudformation_client, changeset_prefix="samcli-deploy"):
         self._client = cloudformation_client
         self.changeset_prefix = changeset_prefix
-        # 500ms of sleep time between stack checks and describe stack events.
-        self.client_sleep = 0.5
+        try:
+            self.client_sleep = int(os.getenv("SAM_CLI_DEPLOY_DESCRIBE_STACK_INTERVAL", str(DEFAULT_CLIENT_SLEEP)))
```

Comment:
This seems like a good change, however we don't use this pattern (of reading from env var) to control command options. We read from env var for very few things like SAM_DEV, REGION, CREDS but not to configure command. Another problem with using env var is that this option is now hidden in the code and is difficult for users to find. If we do it as a command option, it'll be more helpful and easily discoverable.

Thoughts?

Reply:
Is was asked to be an env var (https://github.com/aws/aws-sam-cli/pull/2748#discussion_r604445878) because another PR did that using a command line argument but and got abandoned. Also if this is accepted the documentation could be updated to document this var.

Reply:
I'm also leaning towards a command line option since for this specific value it would be easier for user to acknowledge (help text and documentation wise). This PR overall looks good to me, in terms of design I will engage a conversation within the team to see if we have a consensus on if env vars or options are preferrable! Would love to help push through this PR